### DISCUSSION TITLE
Write an helper function to replace gtk_hl_set_border_width()

### DISCUSSION THREAD

**Discussion by vmagnin** [vmagnin/gtk-fortran#239] [Ideas]
In GTK 3, it was possible to write:
```fortran
    call gtk_container_set_border_width(win, border)
```
In GTK 4, you must write:
```fortran
      call gtk_widget_set_margin_start (win, border)
      call gtk_widget_set_margin_end (win, border)
      call gtk_widget_set_margin_top (win, border)
      call gtk_widget_set_margin_bottom (win, border)
```
It could be interesting to wrote a helper function in `src/gtk-sup.f90`. Its name should indicate that it is not part of GTK 4, so `gtk_container_set_border_width()` is not a good option.
What about simply `container_set_border_width()` ?

**Comment by jtappin** [comment]
Hi Vincent
Maybe `hl_gtk_widget_set_border_width`, and could there be a `gtk_hl_widget.f90` with routines like that for generic (or near generic) widgets.

**Comment by vmagnin** [comment]
> Hi Vincent
> Maybe `hl_gtk_widget_set_border_width`, and could there be a `gtk_hl_widget.f90` with routines like that for generic (or near generic) widgets.

Hi James,
it is an idea.
If you have received our Slack invitation, you can join us. We are having daily discussions with Jerry and Jens about gtk-fortran (or other stuffs) there.
